### PR TITLE
Add network interface label to network metrics

### PR DESCRIPTION
**Pull request by james-bebbington**
Add network interface label to network metrics & added config option to filter by interface name

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_metadata.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 // network metric constants
 
 const (
+	deviceLabelName    = "device"
```
Comment (asuresh4) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_metadata.go:24]: Maybe this label could be called `interface` instead? That way it's consistent with the config option.

**Comment by james-bebbington** [comment]
Updated README

File: receiver/hostmetricsreceiver/README.md [hunk 2]
```diff
@@ -38,3 +44,56 @@ service:
     metrics:
       receivers: [hostmetrics, hostmetrics/disk]
 ```
+
+## Scrapers
+
+The available scrapers are:
```
Comment (nilebox) [inline receiver/hostmetricsreceiver/README.md:50]: nit: the corresponding scraper names in the config are not documented, e.g. 
```markdown
- **CPU** (`cpu`): ...
```

Might be easier to read if converted to table.

File: receiver/hostmetricsreceiver/README.md [hunk 3]
```diff
@@ -38,3 +44,56 @@ service:
...
+- **Process** (Linux & Windows only): Per process CPU, Memory, and Disk I/O metrics
+
+Several scrapers support additional configuration:
+
+#### Disk
+
+```yaml
+disk:
+  <include|exclude>:
+    devices: [ <device name> ... ]
```
Comment (nilebox) [inline receiver/hostmetricsreceiver/README.md:69]:
```suggestion
    devices: [ <device name>, ... ]
```

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/config.go [hunk 4]
```diff
@@ -14,9 +14,24 @@
...
+	"go.opentelemetry.io/collector/internal/processor/filterset"
+	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
+)
 
 // Config relating to Network Metric Scraper.
 type Config struct {
 	internal.ConfigSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// Include specifies a filter on the network interfaces that should be included from the generated metrics.
+	// Exclude specifies a filter on the network interfaces that should be excluded from the generated metrics.
```
Comment (nilebox) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/config.go:27]: This comment should be next to `Exclude` field, and the line below will need to be duplicated probably.

**Pull request merged**